File: sbe/binance/spot_3_2/generated-python/spot_sbe/ExchangeMaxNumOrdersFilter.py
"""Generated SBE (Simple Binary Encoding) message codec."""

import struct
from typing import List, Optional, Tuple
from io import BytesIO

class ExchangeMaxNumOrdersFilter:
    """SBE message: ExchangeMaxNumOrdersFilter."""

    TEMPLATE_ID = 15
    SCHEMA_ID = 3
    SCHEMA_VERSION = 2
    BLOCK_LENGTH = 8

    def __init__(self):
        self.filter_type = None
        self.max_num_orders = None

    def encode(self) -> bytes:
        """Encode the message to bytes."""
        buffer = BytesIO()

        if self.max_num_orders is not None:
            buffer.write(struct.pack('<q', self.max_num_orders))

        return buffer.getvalue()

    def decode(self, data: bytes) -> None:
        """Decode the message from bytes."""
        buffer = BytesIO(data)

        self.max_num_orders = struct.unpack('<q', buffer.read(8))[0]

File: sbe/binance/spot_3_2/generated-php/spot_sbe/ExchangeMaxNumOrdersFilter.php
<?php
/**
 * Generated SBE (Simple Binary Encoding) message codec.
 */

class ExchangeMaxNumOrdersFilter
{
    public const TEMPLATE_ID = 15;
    public const SCHEMA_ID = 3;
    public const SCHEMA_VERSION = 2;
    public const BLOCK_LENGTH = 8;

    public int|float|array|null $filterType = null;
    public int|float|array|null $maxNumOrders = null;

    public function encode(): string
    {
        $buffer = '';

        if ($this->maxNumOrders !== null) {
            $buffer .= pack('q', $this->maxNumOrders);
        }

        return $buffer;
    }

    public function decode(string $data): void
    {
        $offset = 0;

        $this->maxNumOrders = unpack('q', substr($data, $offset, 8))[1];
        $offset += 8;

        // Skip to end of block for forward compatibility
        $offset = 8;

    }
}
